x/validator/handler_test.go

```diff
@@ -177,3 +186,360 @@ func queryValidator(setup TestSetup, owner string) (*types.Validator, error) {
+	account3 := dclauthtypes.NewAccount(ba3,
+		dclauthtypes.AccountRoles{dclauthtypes.NodeAdmin, dclauthtypes.CertificationCenter, dclauthtypes.Vendor}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account3)
+
+	valAddr := sdk.ValAddress(testconstants.ValidatorAddress1)
+
+	// propose new disablevalidator
+	msgProposeDisableValidator1 := NewMsgProposeDisableValidator(account1.GetAddress(), valAddr)
+	_, err := setup.Handler(setup.Ctx, msgProposeDisableValidator1)
+	require.NoError(t, err)
+
+	_, isFound := setup.ValidatorKeeper.GetProposedDisableValidator(setup.Ctx, msgProposeDisableValidator1.Address)
+	require.False(t, isFound)
+
+	disabledValidator, isFound := setup.ValidatorKeeper.GetDisabledValidator(setup.Ctx, msgProposeDisableValidator1.Address)
+	require.True(t, isFound)
+	require.Equal(t, msgProposeDisableValidator1.Address, disabledValidator.Address)
+	require.Equal(t, msgProposeDisableValidator1.Creator, disabledValidator.Creator)
+	require.Equal(t, msgProposeDisableValidator1.Creator, disabledValidator.Approvals[0].Address)
+	require.Equal(t, msgProposeDisableValidator1.Info, disabledValidator.Approvals[0].Info)
+	require.Equal(t, msgProposeDisableValidator1.Time, disabledValidator.Approvals[0].Time)
+	require.False(t, disabledValidator.DisabledByNodeAdmin)
+
+	validator, isFound := setup.ValidatorKeeper.GetValidator(setup.Ctx, valAddr)
+	require.True(t, isFound)
+	require.True(t, validator.Jailed)
+}
+
+func TestHandler_OnlyTrusteeCanApproveDisableValidator(t *testing.T) {
+	setup := Setup(t)
+
+	// create Trustees
+	ba1 := authtypes.NewBaseAccount(testconstants.Address1, testconstants.PubKey1, 0, 0)
+	account1 := dclauthtypes.NewAccount(ba1,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID1)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account1)
+
+	ba2 := authtypes.NewBaseAccount(testconstants.Address2, testconstants.PubKey2, 0, 0)
+	account2 := dclauthtypes.NewAccount(ba2,
+		dclauthtypes.AccountRoles{dclauthtypes.NodeAdmin, dclauthtypes.CertificationCenter, dclauthtypes.Vendor}, nil, testconstants.VendorID2)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account2)
+
+	ba3 := authtypes.NewBaseAccount(testconstants.Address3, testconstants.PubKey3, 0, 0)
+	account3 := dclauthtypes.NewAccount(ba3,
+		dclauthtypes.AccountRoles{dclauthtypes.Trustee}, nil, testconstants.VendorID3)
+	setup.DclauthKeeper.SetAccount(setup.Ctx, account3)
+
+	// propose and approve new disablevalidator
+	msgProposeDisableValidator := NewMsgProposeDisableValidator(account1.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+	_, err := setup.Handler(setup.Ctx, msgProposeDisableValidator)
+	require.NoError(t, err)
+
+	msgApproveDisableValidator := NewMsgApproveDisableValidator(account2.GetAddress(), sdk.ValAddress(testconstants.ValidatorAddress1))
+	_, err = setup.Handler(setup.Ctx, msgApproveDisableValidator)
```

Comment:
Validator disablement was approved earlier - on the proposal step - because the total number of trustees is 2 and so the necessary votes count is 0.66 * 2 ~ 1. So the test scenario must be corrected to correspond to the test name.